### PR TITLE
Add Luci for Eoip package

### PR DESCRIPTION
**Pull request by bogdik**
For https://github.com/openwrt/openwrt/pull/3721

### CODE REVIEW

**Comment by stangri** [comment]
Hey, thanks for the contribution. There was a discussion if the new lua-based luci packages will be accepted or not (since the transition to the javascript-based WebUI), I hope @hnyman or @Ansuel  or @feckert will chime in on that.

There are still a few things which need to be fixed to improve chances that this will be merged:
1. Please add "luci-app-eoip: " prefix to the commit message.
2. Please squash all commits into one commit and force-push the changes.
3. Please sign the commit with your real name.

**Comment by hnyman** [comment]
Reference to packages PR https://github.com/openwrt/packages/pull/14334

Regarding LuCI, 
* there should be just one commit
* the commit needs to be properly titled, and signed off by you
* .pot file is empty although you have created
* preferably the new packages should use javascript instead of lua. This is so small GUI package, that it should be easy to convert.

**Comment by bogdik** [comment]
Done

**Comment by bogdik** [comment]
On last add info for dybamic option

**Comment by bogdik** [comment]
> * there should be just one commit
> * the commit needs to be properly titled, and signed off by you
> * .pot file is empty although you have created
> * preferably the new packages should use javascript instead of lua. This is so small GUI package, that it should be easy to convert.

Check please

File: applications/luci-app-eoip/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2020 Bogdan Shatik (bogdikxxx@mail.ru)
+# This is free software, licensed under the Apache License, Version 2.0
+
+include $(TOPDIR)/rules.mk
+
+PKG_MAINTAINER:=Bogdan Shatik <bogdikxxx@mail.ru>
+PKG_NAME:=luci-app-eoip
+PKG_VERSION:=1.0.1
```
Comment (hnyman) [inline applications/luci-app-eoip/Makefile:8]: There is no need for versioning in the LuCI packages as there is automatic versioning based on commit dates.
You can remove the PKG_VERSION line.

File: applications/luci-app-eoip/htdocs/luci-static/resources/view/eoip/eoip.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		
+		d = s.option(form.Value, "dst", _("Destination"), _("Destination IP address for connection EoIP."));
+		d.nocreate = true;
+		d.datatype = "ipaddr";
+		d.placeholder = "0.0.0.0";
+		
+		o = s.option(form.Flag, "dynamic", _("Dynamic"), _("If you use dynamic option, take attention that there is no authorization, and it is not secure. It is not good idea to use this feature with public ip or insecure(not completely under your control, each host) network."));
+		
+		v = s.option(form.DynamicList,"vlan", _("VLAN"), _("VLAN TAG on this interface"));
+		v.datatype = "and(and(max(4094), integer),and(min(1), integer))";
```
Comment (hnyman) [inline applications/luci-app-eoip/htdocs/luci-static/resources/view/eoip/eoip.js:35]: "range" datatype would be much easier and more clear.
`range(1,4094)`

File: applications/luci-app-eoip/Makefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (hnyman) [inline applications/luci-app-eoip/Makefile:9]: You might add the target-indenpendent line:
`LUCI_PKGARCH:=all`

Comment (hnyman) [inline applications/luci-app-eoip/Makefile:9]: You should add the license database line. Your copyright sets is as Apache 2

`PKG_LICENSE:=Apache-2.0`

**Review by hnyman — CHANGES_REQUESTED** [review]
Some small comments

**Comment by hnyman** [comment]
I have one more comment:

I think that there are still invalid datatypes:   `"min(0), integer"` looks faulty, used in two places.
It should be `"and(min(0), integer)"`  or likely it could just be the unsigned integer type `"uinteger", if the only restriction is that it needs to be non-negative integer. that "uinteger" pretty much defined that by itself.

Have you tested the app?

**Comment by bogdik** [comment]
> Have you tested the app?

Yes, tested fine

**Comment by bogdik** [comment]
Compiled, tested, everything is fine, everything works

**Pull request merged**